python/scripts/process_osm.py

```diff
@@ -74,25 +84,56 @@ def run_osm2pgsql_cmd(config: CredentialsConfig, input_file: str, style_file_pat
         logger.debug(f"Begin importing with: '{' '.join(cmd)}'")
     else:
         logger.info(f"Begin importing with: '{' '.join(cmd)}'")
-    res = subprocess.run(cmd)
-    if not res.returncode:
+    res = subprocess.run(cmd).returncode
+    if not res:
         logger.info("Importing completed.")
+    else:
+        logger.error(f"Error during import: {res}")
+
+def postprocess_osm_import(config: CredentialsConfig, style_file_path: str, schema: str) -> int:
+    """Applies postprocessing SQL associated with **style_file_path** to data in **schema** after importing.
+    """
+    style_file_path = os.path.basename(style_file_path)
+    
+    if style_file_path not in POSTPROCESS_DICT:
+        logger.warning(f"No post-processing defined for style {style_file_path}")
+        return 0
+
+    sql_file_path = str(SQL_DIR / POSTPROCESS_DICT[style_file_path])
+    command = ["psql", "-d", config.db_name, "-U", config.username, "-h", config.db_host, "-p", 
+               str(config.db_server_port), "-c", f"SET search_path TO {schema};", "-f", sql_file_path]
 
-def import_osm_to_db(input_file: str, force: bool, style_file_path: str = None, schema: str = "public") -> int:
-    """Return the size of OSM file in bytes if file found and imports OSM file do database specified in config.ini file.
+    logger.info("Post-processing OSM data after import...")
+    res = subprocess.run(command).returncode
 
-    The **default.lua** style file is used if not specified or set otherwise.
+    if res != 0:
+        logger.error(f"Error during post-processing: {res}")
+        return res
```

Comment:
Do not use return codes as a return value in high-level languages (anything higher than C). If the behavior is not to be expected and recoverable raise an exception, otherwise use e.g. enum Ok, Err

Reply:
The function I used here was sampled from another [branch ](https://github.com/aicenter/road-graph-tool/blob/pipeline-integration-testing/python/scripts/process_osm.py#L83). I admit I didn't concern myself with how the output was used there further. I wanted to preserve that. But you're right that it's not good practice so I removed that and instead handled the exceptions with regular try...except.